### PR TITLE
Joneskm/anyhow

### PR DESCRIPTION
**Pull request by joneskm**
The start of deprecating `AppError`

### CODE REVIEW

File: gaia-rs/src/abci_handler.rs [hunk 1]
```diff
@@ -158,25 +159,25 @@ impl ABCIHandler for GaiaABCIHandler {
         &self,
         ctx: &QueryContext<DB, GaiaStoreKey>,
         query: RequestQuery,
-    ) -> Result<bytes::Bytes, AppError> {
+    ) -> Result<bytes::Bytes, anyhow::Error> {
```
Comment (NYBACHOK) [inline gaia-rs/src/abci_handler.rs:162]: Why no anyhow::Result?

File: gaia-rs/src/abci_handler.rs [hunk 2]
```diff
@@ -158,25 +159,25 @@ impl ABCIHandler for GaiaABCIHandler {
...
+            Err(anyhow!("query path not found"))
         }
     }
 
     fn run_ante_checks<DB: Database>(
         &self,
         ctx: &mut TxContext<'_, DB, GaiaStoreKey>,
         tx: &TxWithRaw<Message>,
-    ) -> Result<(), AppError> {
+    ) -> Result<(), anyhow::Error> {
```
Comment (NYBACHOK) [inline gaia-rs/src/abci_handler.rs:180]: Same here

**Review by NYBACHOK — APPROVED** [review]

**Comment by NYBACHOK** [comment]
There is something which you could do for our `RunTxError` and others. Add variant with `anyhow::Error`. Not sure how to describe it and why it may be useful

File: gaia-rs/src/abci_handler.rs (hunk 1 above)
Comment (joneskm) [reply gaia-rs/src/abci_handler.rs]: Good point. I've switched to anyhow::Result.

**Comment by joneskm** [comment]
> There is something which you could do for our `RunTxError` and others. Add variant with `anyhow::Error`. Not sure how to describe it and why it may be useful

I've replaced the `RunTxError::Custom` variant with an `Application` variant which wraps an anyhow error.

**Review by dmvict — APPROVED** [review]

**Review by NYBACHOK — APPROVED** [review]

**Comment by joneskm** [comment]
Closed in favour of https://github.com/rumos-io/gears/pull/199

**Pull request closed without merging**